### PR TITLE
overwrite this with the version from CABLE-POP-TRENDY

### LINKED ISSUE #251: CABLE-POP_TRENDY refactor: core/biogeophys/cable_climate.F90

### PR DESCRIPTION
**Pull request by JhanSrbinovsky**
# CABLE

Thank you for submitting a pull request to the CABLE Project.

## Description

This keeps getting in the way of merging AND I dont think anyone else uses this SUBROUTINE? Hence requesting review by Ian

Fixes #251

## Checklist

- [X ] The new content is accessible and located in the appropriate section.
- [ X] I have checked that links are valid and point to the intended content.
- [ X] I have checked my code/text and corrected any misspellings 
                 - ALTHOUGH this is as it appears in CABLE-POP-TRENDY. 
                 - Tidying and ensuring compliance at this stage would make the move redundant

----
📚 Documentation preview 📚: https://cable--378.org.readthedocs.build/en/378/

### CODE REVIEW

File: src/science/misc/cable_climate.F90 [hunk 1]
```diff
@@ -15,271 +15,475 @@
...
-    IF(MOD(ktau,ktauday)==1) THEN
-       climate%dtemp = met%tk - 273.15
-       climate%dmoist = canopy%fwsoil
-    ELSE
-       climate%dtemp = climate%dtemp + met%tk - 273.15
-       climate%dmoist = climate%dmoist + canopy%fwsoil
-    ENDIF
+    if (mod(ktau, ktauday) == 1) then
+       climate%dtemp     = met%tk - 273.15
+       climate%dmoist    = sum(real(ssnow%wb(:,:))*veg%froot(:,:), 2)
```
Comment (har917) [inline src/science/misc/cable_climate.F90:101]: I think we need to retain a comment here around the change in science from `%dmoist = %fwsoil` to `%dmoist  = SUM(%wb * %froot, 2)`.  These quantities are related but different.

File: src/science/misc/cable_climate.F90 [hunk 2]
```diff
@@ -15,271 +15,475 @@
...
 
+       ! Update GDD counters and chill day count
+       climate%gdd0  = climate%gdd0  + max(0.0, climate%dtemp-0.0)
+       climate%agdd0 = climate%agdd0 + max(0.0, climate%dtemp-0.0)
 
-       ENDIF
+       climate%gdd5  = climate%gdd5  + max(0.0, climate%dtemp-5.0)
+       climate%agdd5 = climate%agdd5 + max(0.0, climate%dtemp-5.0)
 
-       !jhan: see CABLE Ticket#149 and above CABLE_LSM: comment
```
Comment (har917) [inline src/science/misc/cable_climate.F90:146]: May need to retain this `#ifndef` condition - I don't know why it was necessary and likely superceded by the  `#ifdef UM_CBL`

File: src/science/misc/cable_climate.F90 [hunk 3]
```diff
@@ -330,223 +536,230 @@ SUBROUTINE biome1_pft(climate, np)
...
        ! (17) Ice/polar desert
-       IF(climate%biome(k)==999) THEN
+       if (climate%biome(k)==999) then
           climate%biome(k) = 17
           climate%iveg(k) = 17  !
-       ENDIF
+       endif
 
-       !jhan: see CABLE Ticket#149 and above CABLE_LSM: comment
-# ifndef ESM15
```
Comment (har917) [inline src/science/misc/cable_climate.F90:543]: see earlier comment about this `#ifndef`

File: src/science/misc/cable_climate.F90 (hunk 1 above)
Comment (JhanSrbinovsky) [reply src/science/misc/cable_climate.F90:101]: such as?

OK before we get too involved, I am. totally unfamiliar with this module - who uses cable_climate?

File: src/science/misc/cable_climate.F90 [hunk 4]
```diff
@@ -559,544 +772,171 @@ SUBROUTINE biome1_pft(climate, np)
...
+
+    !if (.not.cable_user%climate_fromzero) then
+    !   CALL READ_CLIMATE_RESTART_NC (climate, ktauday)
     !endif
 
-  END SUBROUTINE climate_init
-
-  ! ==============================================================================
-
-  SUBROUTINE WRITE_CLIMATE_RESTART_NC ( climate, ktauday )
```
Comment (har917) [inline src/science/misc/cable_climate.F90:634]: Where have the `write_climate_restart_nc` and `read_climate_restart_nc` subroutines/capability gone?  Ah-ha possibly into `write_netcdf_cbm_var()` which is in `cable_def_types_mod`.

So this PR may need to deal with that MODULE as well.

File: src/science/misc/cable_climate.F90 (hunk 1 above)
Comment (har917) [reply src/science/misc/cable_climate.F90:101]: As far as I know only CABLE-POP simulations actively use this routine - so that's the TRENDY and BIOS applications - however the `climate%` is woven throughout the offline code.

I'm not actually sure how CM2 and AM3 are handling this to be honest - the TYPE is there (if incomplete) and there is an allocation/initialisation.  However there is no call to the `cable_climate()` subroutine yet there is a use (in AM3) in `casa_rplant` - I would have thought that would have caused some 'interesting' results or a crash.

Comment (har917) [reply src/science/misc/cable_climate.F90:101]: To answer the implicit question - all we need to do is include a comment such as
`! Has been updated from %dmoist  = canopy%fwsoil `  with, if possible, a svn revision number.

Comment (har917) [reply src/science/misc/cable_climate.F90:101]: > there is a use (in AM3) in casa_rplant and `cbl_dryLeaf`

Specifically `casa_rplant()` and `cbl_dryLeaf` use `climate%qtemp_max_last_year` - this is initialised to 0.0.  HOwever both uses are inside a IF `cable_user%CALL_climate` condition so will only be active if triggered by the user.

File: src/science/misc/cable_climate.F90 [hunk 5]
```diff
@@ -15,271 +15,475 @@
 ! Called from: cable_driver
 !
 ! History: Vanessa Haverd Jan 2015
-
+! Aug 2017: additional leaf-level met variables stored for use in optimising ratio
+! of Jmax to Vcmax
+#ifndef UM_CBL
```
Comment (har917) [inline src/science/misc/cable_climate.F90:20]: mmm - not sure I really like this in the code base at least not without the corresponding #ifdef case

**Review by har917 — COMMENTED** [review]
I think this is okay to go in with some key notes:
1. Care will be needed around the `#ifdef`s
2. I suspect that this will need partner updates to cable_def_types_mod() to work. 
3. this is one of the routines that will need to be rationalised as part of CABLE-4 - i.e. this PR should be viewed only as a temporary fix.

@ccarouge in light of the above - are you happy that @JhanSrbinovsky goes ahead and merges into MAIN (assuming that any technical quirks can be overcome)?

**Comment by JhanSrbinovsky** [comment]
the ifndef um_cbl was because it is NEVER used in any UM application and there were corresponding things to bring in at build time which would have complicated things for a subroutine that isnt even called. That said - the version from CABLE-POP is even worse. They've added things to climate% TYPE which is I suppose expected. They've even added elements to met% TYPE.  

Online, climate% TYPE is declared etc, they are just never used in calculation of anything (well they shouldn't be anyway, I hope this is the case).


There will be more work needed to even get this to build properly! I would be tempted to just abandon this altogether, except that it needs to go in if indeed there is to be a CABLE4 code base. 

At the very least there will have to be things "taken out" (or reviewed if they are even necessary) that apply to reading and writing netcdf files.

In summary, 

- work needs to be done for this cable_climate to even be substituted to offline CABLE, including for applications where it is not called.  
- further thinking into what role this would play in an online CABLE4 is required. 

Close this PR until these issues are resolved
